src/test/java/use_cases/review_use_case/ReviewInteractorTest.java

```diff
@@ -0,0 +1,96 @@
+package use_cases.review_use_case;
+
+import entities.Review;
+import org.junit.jupiter.api.Test;
+import screens.review_screen.IReviewView;
+import screens.review_screen.ReviewTestScreen;
+import use_cases.regular_user_register_use_case.*;
+import screens.review_screen.InMemoryReview;
+
+import javax.management.InvalidAttributeValueException;
+import java.io.IOException;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class ReviewInteractorTest {
+
+    @Test
+    void addReview() throws IOException, InvalidAttributeValueException {
+        ReviewGateway reviewRepository = new InMemoryReview();
+        IReviewView iView =  new ReviewTestScreen();
+        ReviewOutputBoundary presenter = new ReviewPresenter(iView) {
+            @Override
+            public ReviewResponseModel reportReview(ReviewResponseModel review) {
+                assertEquals("added", review.getStatus());
+                assertNotNull(review.getCreationTime());
+                return null;
+            }
+        };
+
+        UserGateway userGateway = new UserGateway(System.getProperty("user.dir"));
+        ReviewInputBoundary interactor = new ReviewInteractor(
+                presenter, reviewRepository, userGateway);
+
+        UserRegisterDsRequestModel userRegisterDsRequestModel = new UserRegisterDsRequestModel("Bob", "password", "password",
+                null, null, null, null, null, null, null, null, null);
```

Comment:
Many repeated nulls here (just checking if they are intended).

Reply:
they are just me being lazy about constructing the request model since it seems to allow for null input. I can input the "real" information in it if this is confusing.

Reply:
Okay, thanks for the clarification, and I don't think changing it would be necessary!